ADDING_A_PLUGIN.md

```diff
@@ -0,0 +1,347 @@
+  metadata_content = '{}'  # We have no metadata to store.
+  summary_metadata.plugin_data.add(plugin_name=PLUGIN_NAME,
+                                   content=metadata_content)
+
+  summary = tf.Summary()
+  summary.value.add(tag=tag,
+                    metadata=summary_metadata,
+                    tensor=tensor)
+  return summary
+```
+
+The `op` and`pb` methods above are two different ways of creating the same data: `Summary` protobufs containing `"Hello, $guest"` as a string tensor, with a unique `tag`, and with the appropriate metadata.
+
+We've discussed above why we need the `tag` (to differentiate data series) and the `tensor` (to hold the actual payload data). But why do we need metadata?
+
+Without the metadata, there is no way for plugins to know which `Summary`s they are interested in. How would the `Greeter` plugin know that these `Summary`s weren't intended for a different plugin that also handles text data, and just happened to have "Hello" in the beginning of the message?
+
+The `Greeter` plugin's summaries are distinguished by the fact that they all have a `plugin_data` entry in the `SummaryMetadata` which is marked with the `Greeter` `plugin_name`, which is, appropriately, `"greeter"`.
+
+The `SummaryMetadata` can also contain additional information describing each series. TensorBoard uses it to store the `display_name` and `description`. These both describe the series of data as a whole, rather than individual data points, so it makes sense  to put them in the metadata. Your plugin could also define some additional metadata - that can go in the `metadata_content` field. Traditionally, if you have `metadata_content`, you should put in either
+- a custom protocol buffer for your plugin, serialized as a ByteString or String
+- a JSON object, serialized as a string
+
+The metadata may be quite long - imagine writing a long description (with hyperlinks, markdown code blocks, et cetera) which explains exactly how a certain data series is calculated in your model. To save space, we don't write the metadata every time. The `tf.summary.FileWriter` only writes metadata once per tag. Thus, changing the metadata for a given tag is a bad idea.
+
+You might be wondering why we have `tag` and `display_name` as separate concepts. Why not always use the `tag` as the display name in TensorBoard?
+
+This happens because we can't allow tags to collide. If tag collision happens, it's impossible to tell which data came from which series. TensorFlow prevents tag collision, by using the summary op's node name as the tag; that means it is guaranteed to be unique by the TensorFlow naming system. But, that also means implementation noise from the graph is included in a tag. If you add a scalar op with the name `"loss"` deep inside your graph, its tag might become something like `"tower1/layer8/loss"`. Keeping the display name as a separate concept ensures that your TensorBoard will still be easy to read.
```

Comment:
english: remove commas in "TensorFlow prevents tag collision, by using the summary op's node name as the tag" and in "But, that also means…"